atc/component/scheduler.go

```diff
@@ -0,0 +1,76 @@
+package component
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+)
+
+var Clock = clock.NewClock()
+
+type NotificationsBus interface {
+	Listen(string) (chan bool, error)
+	Unlisten(string, chan bool) error
+}
+
+// Schedulable represents a workload that is executed normally on a periodic
+// schedule, but can also be run immediately.
+type Schedulable interface {
+	RunPeriodically(context.Context)
+	RunImmediately(context.Context)
+}
+
+// Scheduler runs a workload periodically, or immediately upon receiving a
+// notification.
+type Scheduler struct {
```

Comment:
Calling this a `Scheduler` is a bit confusing given the importance of the actual scheduler. Maybe just simply `Runner`?

Reply:
Yeah agreed, this is one of those wishy-washy names and we already use it elsewhere. I guess `Runner` gels well with `Runnable`. I can try that out!